app/Services/ArticleLinkSuggestionService.php

```
@@ -146,15 +147,16 @@ public function analyzeForSource(Article $source): Collection
-                $results->push(ArticleLinkSuggestion::create($attributes + [
-                    'source_article_id' => $source->id,
-                    'target_article_id' => $candidate->id,
-                ]));
-            }
+            // updateOrCreate (non un controllo di esistenza seguito da un
+            // create separato) resta corretto anche se due richieste
+            // "Analizza" concorrenti per lo stesso articolo colpiscono la
+            // stessa coppia sorgente/target: senza questo, un doppio click
+            // potrebbe far scattare il vincolo unique(source,target) e far
+            // fallire la richiesta con un errore 500.
+            $results->push(ArticleLinkSuggestion::updateOrCreate(
```

PHPStan (2.2.7)

[error] Call to an undefined static method App\Models\ArticleLinkSuggestion::updateOrCreate().


(staticMethod.notFound)